### PR TITLE
More DApp3 fixes

### LINKED ISSUE #2608: Update confirm transaction modal for our mobile app when transaction is meta transaction
Modal asks mobile user to still confirm transaction even when it is being handled automatically by meta transactions.

Simulator Screen Shot - iPhone Xʀ - 2019-07-03 at 14 48 33

### PR DESCRIPTION
**Pull request by shahthepro**
[Link to Test Build](https://originprotocol.github.io/test-builds/shah/#/)

### Description:
- Show back link and search bar when user navigates to a sub-route
- Show Hamburger menu when profile page is accessed directly; otherwise show a back icon
- Add "No reviews" image and text
- Add "More" and "Feeback" links to Hamburger menu
- Redirect user to the conversation after sending a message from modal
- Load default conversation when `/messages` is accessed directly
- Show a "Waiting for confirmation" message when meta transaction is enabled and user is on Origin Wallet app (Closes #2608)
- Remove unnecessary spacing below the search popover.

### CODE REVIEW

File: dapps/marketplace/src/pages/listings/_EarnTokensCard.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+const EarnTokensCard = ({ wallet }) => {
+  return (
+    <Query
+      query={enrollmentStatusQuery}
+      variables={{
+        walletAddress: wallet || '0xdummyAddress'
+      }}
+      // enrollment info can change, do not cache it
+      fetchPolicy="network-only"
```
Comment (sparrowDom) [inline dapps/marketplace/src/pages/listings/_EarnTokensCard.js]: This is very nitpicky of me. You can simplify this class and have the Link always point to `/campaigns`. This way you don't need to query `enrollmentStatus` and if the user is not signed into the Rewards program he will be automatically redirected to `/welcome`

Comment (shahthepro) [reply dapps/marketplace/src/pages/listings/_EarnTokensCard.js]: Thanks @sparrowDom. Have updated it.

**Comment by nick** [comment]
A few conflicts here now since I merged the CTA PR

**Comment by shahthepro** [comment]
@nick Have resolved the merge conflict

**Pull request merged**